### PR TITLE
Give game makers more control

### PR DESCRIPTION
**Pull request by phrohdoh**
This adds to the `Info.plist`:
 - The string `{BUNDLE_DISPLAY_NAME}` to be replaced (via `sed`, or `plutil`, etc) by the game's title (this could be renamed `GAME_DISPLAY_NAME` or even `GAME_NAME` if desired)
 - The string `{DEFAULT_MOD_ID}` which would (as an example for RA2) be replaced with `ra2` (this defaults to ~~`modchooser`~~ `nil` and if not supplied `Game.Mod` and `Engine.DefaultMod` will not be passed)

Changes to the build system:
 - Moved output assemblies into a `build` directory which the `clean` target removes

### CODE REVIEW

File: OpenRA.m [hunk 1]
```diff
@@ -121,15 +121,20 @@ - (void)launchGameWithArgs: (NSArray *)gameArgs
 
 	launched = YES;
 
-	// Default values - can be overriden by setting MonoMinVersion and MonoGameExe in Info.plist
+	// Default values - can be overriden by setting certain keys Info.plist
 	NSString *gameName = @"OpenRA.Game.exe";
+	NSString *defaultModId = @"modchooser";
```
Comment (pchote) [inline OpenRA.m]: Hardcoding this to modchooser isn't great considering that we're about to kill that.  Can you please default this to `nil`, and then only add the launch arguments if it is non-nil?

Comment (phrohdoh) [reply OpenRA.m]: Done.

**Comment by phrohdoh** [comment]
Updated.

File: Makefile [hunk 2]
```diff
@@ -1,22 +1,28 @@
...
 
 OpenRA: OpenRA.m
-	clang -m32 OpenRA.m -o OpenRA -framework AppKit -mmacosx-version-min=10.6
+	@mkdir -p build
+	@clang -m32 OpenRA.m -o build/OpenRA -framework AppKit -mmacosx-version-min=10.6
 
 launchgame: launchgame.m
-	clang -m32 launchgame.m -o launchgame -framework AppKit -mmacosx-version-min=10.6
+	@mkdir -p build
+	@clang -m32 launchgame.m -o build/launchgame -framework AppKit -mmacosx-version-min=10.6
```
Comment (pchote) [inline Makefile]: I'm not a fan of suppressing *all* output from the makefile steps.  Can you please print out "Compiling launchgame", "Compiling OpenRA", "Compressing launcher.zip" etc?

File: OpenRA.m [hunk 3]
```diff
@@ -121,15 +121,20 @@ - (void)launchGameWithArgs: (NSArray *)gameArgs
...
+	NSString *defaultModId = nil;
 
 	NSDictionary *plist = [[NSBundle mainBundle] infoDictionary];
 	if (plist)
 	{
 		NSString *exeValue = [plist objectForKey:@"MonoGameExe"];
 		if (exeValue && [exeValue length] > 0)
 			gameName = exeValue;
+
+		NSString *defaultModIdValue = [plist objectForKey:@"DefaultModId"];
```
Comment (pchote) [inline OpenRA.m]: I've come to the conclusion that our only feasible way forwards on the `OpenRA.Game.exe` side is to make `Game.Mod` a required commandline parameter and remove the problematic `Mod` settings.yaml key that is causing all our problems.  Once that is done `Engine.DefaultMod` will be redundant and can be removed.

Looking ahead to this change, can you please drop the "default" from the plist key and variable names?

File: OpenRA.m [hunk 4]
```diff
@@ -142,6 +147,13 @@ - (void)launchGameWithArgs: (NSArray *)gameArgs
 	[launchArgs addObject: @"--debug"];
 	[launchArgs addObject: [gamePath stringByAppendingPathComponent: gameName]];
 	[launchArgs addObject: [NSString stringWithFormat:@"Engine.LaunchPath=\"%@\"", appPath]];
+
+	if (defaultModId)
+	{
+		[launchArgs addObject: [NSString stringWithFormat:@"Engine.DefaultMod=%@", defaultModId]];
```
Comment (pchote) [inline OpenRA.m]: We can keep this for now, because it will be needed during the transition and won't do any harm after it is complete.

**Comment by pchote** [comment]
Can you please also extract the hardcoded FAQ link at [OpenRA.m#L59](https://github.com/OpenRA/OpenRALauncherOSX/blob/master/OpenRA.m#L59) to an info.plist key?  Third party mods may want to change this to.

File: Info.plist [hunk 5]
```diff
@@ -119,5 +119,9 @@
 			</dict>
 		</dict>
 	</array>
+	<key>DefaultModId</key>
```
Comment (phrohdoh) [inline Info.plist]: I need to rename this too.

**Comment by pchote** [comment]
The launcher needs several additional changes to support the things that I want for our split official mod launchers.  Rather than get you to write what I have in mind by proxy I have taken this as a base, and been working on it at [per-mod-template](https://github.com/OpenRA/OpenRALauncherOSX/compare/per-mod-template?expand=1) in parallel with OpenRA/OpenRA packaging script changes.

If you're up for it I'll finish my changes, add back your makefile changes (plus my review requests), and then push it over this PR for review and merge.

**Comment by phrohdoh** [comment]
Sounds good to me.

**Comment by pchote** [comment]
Ok, done.  This gets a :+1: from me as-is.  If you're happy with my changes then :+1: and merge.

File: Info.plist [hunk 6]
```diff
@@ -5,17 +5,17 @@
...
 	<string>English</string>
 	<key>CFBundleDisplayName</key>
-	<string>OpenRA</string>
+	<string>{MOD_NAME}</string>
 	<key>CFBundleExecutable</key>
 	<string>OpenRA</string>
 	<key>CFBundleIconFile</key>
 	<string>OpenRA.icns</string>
 	<key>CFBundleIdentifier</key>
-	<string>net.openra.launcher</string>
```
Comment (phrohdoh) [inline Info.plist:53]: People will probably want to change this entire value but for the sake of moving this work forward we can do that later.

**Pull request merged**